packages/core/src/magicswap/swap.ts

```diff
@@ -0,0 +1,196 @@
+import { type Config, connect, mock, readContract } from "@wagmi/core";
+import { erc20Abi } from "viem";
+import { erc1155Abi } from "../abis/erc1155Abi";
+import type { AddressString, SupportedChainId } from "../types";
+import { getContractAddresses } from "../utils/contracts";
+import { DEFAULT_WAGMI_CONFIG } from "../utils/wagmi";
+import type { Pool } from "./fetchPools";
+import { getSwapRoute } from "./getSwapRoute";
+import type { PoolToken } from "./types";
+
+type NFTInput = { id: string; quantity: number };
+
+const getAmountMax = (amount: bigint, slippage: number) =>
+  amount + (amount * BigInt(Math.ceil(slippage * 1000))) / 1000n;
+const getAmountMin = (amount: bigint, slippage: number) =>
+  amount - (amount * BigInt(Math.ceil(slippage * 1000))) / 1000n;
+
+const DEFAULT_SLIPPAGE = 0.005;
+
+export const swap = async ({
```

Comment:
The function doesn't actually perform a swap, so should maybe be named something like `createSwapArgs` instead